Dan Tudor/09 - 3.2 - 3.7/main.c

```diff
@@ -0,0 +1,123 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+void substring(char s[30], int n, int m)
+{
+    char sub[30];
+    int i;
+    strncpy(sub,s+n,m);
+    puts(sub);
```

Comment:
Rather ugly way of making sure the result is printed...
Thing is, this function violates a basic principle: it does two things - first, it does what its title says (creates a substring) but then also prints it. Furthermore, I have told you *so. many. times.* not to use static sized arrays/string, and yet, here you are, using statically allocated arrays/strings. Please, don't. You will have to change all these strings of size 30 to dynamically allocated ones.
Thank you

Reply:
Also, make sure to not do the printing in the same function, it's just lazy. Create a function which outputs the result, separately